crt_portal/api/filters.py

```diff
@@ -1,19 +1,96 @@
+# Class to handle filtering data by supplied query params, providing the params are valid.
+
+from cts_forms.models import User, Report, FormLettersSent
 from actstream import registry
 from actstream.models import actor_stream
-from cts_forms.models import User
 import urllib.parse
 from datetime import datetime
 
+from utils.datetime_fns import change_datetime_to_end_of_day
+
+
+def autoresponses_filter(querydict):
+    kwargs = {}
+    autoresponse_qs = Report.objects.filter().all()
+
+    for field in querydict.keys():
+        if "date" in field:
+            # filters by a start date or an end date expects yyyy-mm-dd
+            encoded_date = querydict.getlist(field)[0]
+            decoded_date = urllib.parse.unquote(encoded_date)
+            if field == "start_date":
+                try:
+                    date_obj = datetime.strptime(decoded_date, "%Y-%m-%d")
+                    kwargs["create_date__gte"] = date_obj
+                except ValueError:
+                    # if the date is invalid, we throw an error
+                    raise ValueError("Incorrect date format, should be YYYY-MM-DD")
+            elif field == "end_date":
+                try:
+                    date_obj = datetime.strptime(decoded_date, "%Y-%m-%d")
+                    date_obj = change_datetime_to_end_of_day(date_obj, field)
+                    kwargs["create_date__lte"] = date_obj
+                except ValueError:
+                    # if the date is invalid, we throw an error
+                    raise ValueError("Incorrect date format, should be YYYY-MM-DD")
+        elif field == "assigned_section":
+            kwargs["assigned_section"] = querydict.getlist(field)[0]
+    filtered_autoresponses = autoresponse_qs.filter(**kwargs)
+    return len(filtered_autoresponses)
+
+
+def form_letters_filter(querydict):
+    kwargs = {}
+    form_letters_counter = {}
+
+    form_letters_payload = {
+        "start_date": "",
+        "end_date": "",
+        "assigned_section": "",
+        "total_form_letters": 0,
+        "form_letters_counter": {}
+    }
+
+    section = querydict["assigned_section"]
```

Comment:
This is throwing an exception if you do not have the field on the url.  

<img width="1599" alt="image" src="https://user-images.githubusercontent.com/6232068/168070264-86d096ff-ca8c-465c-8e5f-0584211a1e8e.png">

To recreate, I went to http://localhost:8000/api and clicked the http://localhost:8000/api/form-letters/ link.